src/server/pps/server/pipeline_controller.go

```diff
@@ -278,7 +278,7 @@ func (pc *pipelineController) step(timestamp time.Time) (isDelete bool, retErr e
 	}
 	defer tracing.FinishAnySpan(span, "err", retErr)
 	// derive the latest pipelineInfo with a corresponding auth'd context
-	pipeline := client.NewPipeline(pc.pipeline)
+	pipeline := client.NewPipeline(pc.pipeline.Name)
```

Comment:
don't we need a project here?

Reply:
Honestly, it can just use `pc.pipeline` now.